stats/opentelemetry/grpc_trace_bin_propagator_test.go

```diff
@@ -0,0 +1,253 @@
+		t.Run(test.name, func(t *testing.T) {
+			p := GRPCTraceBinPropagator{}
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+			ctx = oteltrace.ContextWithSpanContext(ctx, test.injectSC)
+			ctx = metadata.NewIncomingContext(ctx, test.incomingMd)
+			ctx = metadata.NewOutgoingContext(ctx, test.outgoingMd)
+
+			c := itracing.NewOutgoingCarrier(ctx)
+			p.Inject(ctx, c)
+
+			if gotKeys := c.Keys(); !cmp.Equal(test.wantKeys, gotKeys, cmpopts.SortSlices(func(a, b string) bool { return a < b })) {
+				t.Errorf("c.Keys() = keys %v, want %v", gotKeys, test.wantKeys)
+			}
+			md, _ := metadata.FromOutgoingContext(c.Context())
+			gotH := md.Get(grpcTraceBinHeaderKey)
+			if !test.injectSC.IsValid() {
+				if len(gotH) > 0 {
+					t.Fatalf("got %v value from Carrier's context metadata grpc-trace-bin header, want empty", gotH)
+				}
+				return
+			}
+			if gotH[len(gotH)-1] == "" {
+				t.Fatalf("got empty value from Carrier's context metadata grpc-trace-bin header, want valid span context: %v", test.injectSC)
+			}
+			gotSC, ok := fromBinary([]byte(gotH[len(gotH)-1]))
+			if !ok {
+				t.Fatalf("got invalid span context %v from Carrier's context metadata grpc-trace-bin header, want valid span context: %v", gotSC, test.injectSC)
+			}
+			if test.injectSC.TraceID() != gotSC.TraceID() && test.injectSC.SpanID() != gotSC.SpanID() && test.injectSC.TraceFlags() != gotSC.TraceFlags() {
+				t.Fatalf("got span context = %v, want span contexts %v", gotSC, test.injectSC)
+			}
+		})
+	}
+}
+
+// TestExtract verifies that the GRPCTraceBinPropagator correctly extracts
+// OpenTelemetry span context data from the provided context using carrier.
+//
+// If a valid span context was injected, it verifies same trace span context
+// is extracted from carrier's metadata for `grpc-trace-bin` header key.
+//
+// If invalid span context was injected, it verifies that valid trace span
+// context is not extracted.
+func (s) TestExtract(t *testing.T) {
+	tests := []struct {
+		name       string
+		wantSC     oteltrace.SpanContext // expected span context from carrier
+		incomingMd metadata.MD
+		outgoingMd metadata.MD
+		wantKeys   []string
+	}{
+		{
+			name:       "valid OpenTelemetry span context",
+			wantSC:     validSpanContext.WithRemote(true),
+			incomingMd: metadata.MD{grpcTraceBinHeaderKey: []string{string(toBinary(validSpanContext.WithRemote(true)))}, "incoming-key": []string{"incoming-value"}},
+			outgoingMd: metadata.MD{"outgoing-key": []string{"outgoing-value"}},
+			wantKeys:   []string{grpcTraceBinHeaderKey, "incoming-key"},
+		},
+		{
+			name:       "invalid OpenTelemetry span context",
+			wantSC:     oteltrace.SpanContext{},
+			incomingMd: metadata.MD{grpcTraceBinHeaderKey: []string{string(toBinary(oteltrace.SpanContext{}))}, "incoming-key": []string{"incoming-value"}},
+			outgoingMd: metadata.MD{"outgoing-key": []string{"outgoing-value"}},
+			wantKeys:   []string{grpcTraceBinHeaderKey, "incoming-key"},
```

Comment:
Keys appears to actually test only the carrier in this instance, right?  So maybe delete this part of the test and make sure it's covered in the carrier tests.